drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -1919,6 +1977,31 @@ D3D12_RENDER_TARGET_VIEW_DESC RenderingDeviceDriverD3D12::_make_rtv_for_texture(
 	return rtv_desc;
 }
 
+D3D12_UNORDERED_ACCESS_VIEW_DESC RenderingDeviceDriverD3D12::_make_ranged_uav_for_texture(const TextureInfo *p_texture_info, uint32_t p_mipmap_offset, uint32_t p_layer_offset, uint32_t p_layers, bool p_add_bases) {
+	D3D12_UNORDERED_ACCESS_VIEW_DESC uav_desc = p_texture_info->aliasing_hack.main_uav_desc;
+
+	// Leveraging union members' compatibility.
```

Comment:
I feel this trick hurts the readability and maintenance of the code. While it's true the compatibility is guaranteed as long as the headers don't change (and they won't), the semantic meaning of the code is lost since it's not accounting for the existence of every view dimension case we want to cover and filling in their corresponding structures. I'd take the time to properly fill out each case here.

Reply:
Sometimes I am lazy... I've reworked it (as well as other spot in the code I was abusing the union) to make the initialization explicit in all cases.